pypesto/optimize/util.py

```diff
@@ -0,0 +1,68 @@
+from ..engine import Engine, SingleCoreEngine
+from ..objective import HistoryOptions
+from ..store.save_to_hdf5 import get_or_create_group
+from pathlib import Path
+
+
+import h5py
+
+
+def check_hdf5_mp(history_options: HistoryOptions,
+                  engine: Engine):
+    """
+    Create a folder for partial HDF5 files,
+    if a parallelization engine will be used.
+
+    Paramters
+    ---------
+    engine:
+        The Engine which is used in the optimization
+    history_options:
+        The HistoryOptions used in the optimization
+
+    Returns
+    -------
+    The filename that will be used to combine the partial HDF5 files later.
+    If a parallelization engine is not used, `None` is returned.
+    """
+    if not isinstance(engine, SingleCoreEngine):
+        filename = history_options.storage_file
+        file_path = Path(filename)
+        # create directory with same name as original file stem
+        partial_file_path = (
+            file_path.parent / file_path.stem /
+            (file_path.stem + '_{id}' + file_path.suffix)
+        )
```

Comment:
would go for
```suggestion
        partial_file_path = os.path.join(
            file_path.parent, <usw ...>
        )
```

Reply:
Also then, you do not need the `str(partial_file_path)`.

Reply:
if I do that, I'd loose the possibility to conveniently create the directory where the files would be stored (line below).